sdk/storage/azure-storage-common/src/test/java/com/azure/storage/common/implementation/BufferStagingAreaTests.java

```diff
@@ -102,4 +106,48 @@ private static Stream<Arguments> bufferStagingAreaSupplier() {
             Arguments.of(100, Constants.MB * 4, Constants.MB * 8) // _
         );
     }
+
+    @ParameterizedTest
+    @MethodSource("bufferStagingAreaWithSmallerBufferSupplier")
+    public void bufferStagingAreaWithSmallerBuffer(int stagingSize, int dataSize) {
+        // Generate random data
+        byte[] bytes = new byte[dataSize];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        Flux<ByteBuffer> byteBufferFlux = Flux.just(buffer);
+
+        BufferStagingArea stagingArea = new BufferStagingArea(stagingSize, stagingSize);
+        List<ByteBuffer> collectedBuffers = byteBufferFlux.flatMapSequential(stagingArea::write, 1)
+            .concatWith(Flux.defer(stagingArea::flush))
+            .flatMap(BufferAggregator::asFlux)
+            .collectList().block();
+
+        assertNotNull(collectedBuffers);
+
+        // Convert list of ByteBuffers to a single byte array
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        collectedBuffers.forEach(bb -> {
+            byte[] array = new byte[bb.remaining()];
+            bb.get(array);
+            try {
+                outputStream.write(array);
```

Comment:
Let's also add an assertion that length of each of these byte buffers is equal to the `stagingSize`. This is an important property that needs to hold for encryption/decryption to work.